packages/flutter_tools/lib/src/build_system/build_system.dart

```diff
@@ -772,9 +772,12 @@ class Node {
       }
     }
 
-    // If we depend on a file that doesnt exist on disk, kill the build.
+    // If we depend on a file that doesnt exist on disk, mark the build as
```

Comment:
What happens if the user listed a file by accident, and the fix is to remove the file? How do they find out?

Reply:
The user here would be the tools team, so ideally we'd have a test to cover it. Long term, I think we should surface the reasons why targets are re-run.

Reply:
Would it be useful to do a `printTrace` here with some diagnostic info?

Reply:
From the description, what is an example of a fairly dynamic rule where a required file doesn't exist at the moment of running the target?

Reply:
Done

Reply:
Anything using a depfile, which will be lots of rules